drivers/i2c/Kconfig.stm32

```diff
@@ -43,4 +43,11 @@ config I2C_STM32_COMBINED_INTERRUPT
 	depends on I2C_STM32_INTERRUPT
 	default y if SOC_SERIES_STM32F0X || SOC_SERIES_STM32G0X || SOC_SERIES_STM32L0X
 
+config I2C_STM32_BUS_RECOVERY
+	bool "Bus recovery support"
+	depends on I2C_STM32_V1 && PINCTRL
```

Comment:
Using I2C requires PINCTRL anyway, so dependecy on PINCTRL is useless.
Then, I wonder about dependency on I2C_STM32_V1. Is that really required ? I don't see anything V1 specific in your proposal.

Reply:
Neither are required - removed in latest push